packages/style-engine/src/index.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { groupBy, kebabCase } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import type { Style, GeneratedCSSRule, StyleDefinition } from './types';
+import { styleDefinitions } from './styles';
+
+/**
+ * Generates a stylesheet for a given style object and selector.
+ *
+ * @param  style    Style object.
+ * @param  selector CSS selector.
+ *
+ * @return generated stylesheet.
+ */
+export function generate( style: Style, selector: string ): string {
```

Comment:
Just thinking, should this signature be a `style` object and an `options` object, instead of the selector being the second param? Then we'd have `selector` as one of the options in the `options` object.

If so, we might do the same for `getCSSRules`.

Reply:
Sounds legit 😆 

Are there further options that we need in these methods, or is it about setting up API for flexibility?

Reply:
I think `options` could make sense here yeah, things like "enabled styles" for instance could be an option, or a flag on whether to optimize or not... 

That said, I'm not too concerned because that package is written as "private" allowing us to make breaking changes before actually considering making it "public".

Reply:
Thanks for the feedback! I thought I'd have a go at it now, so that we don't have to refactor it separately when we go to add another option. I've added an `options` object in https://github.com/WordPress/gutenberg/pull/37978/commits/a925476e0c3d492ec6e6b974949c7bf85be215a7, added an additional type, and pass the object around to the style definitions.

> Are there further options that we need in these methods, or is it about setting up API for flexibility?

Yes, it's about flexibility — we don't need to add additional options _just_ yet for this PR, but we do know that we'd like to explore adding additional ones, so 🤞 switching to `options` now makes it a bit easier for exploratory follow-up PRs.

Let me know if anyone would like the implementation changed, though, of course!